bdb/bdb_verify.c

```diff
@@ -273,245 +213,582 @@ static void printhex(SBUF2 *sb, int (*lua_callback)(void *, const char *),
+                dbt_blob_data.data = NULL;
+
+                rc = bdb_cget_unpack_blob(bdb_state, cblob, &dbt_blob_key,
+                                          &dbt_blob_data, &ver, DB_SET);
+                if (rc == DB_NOTFOUND) {
+                    realblobsz[blobno] = -1;
+                    if (blobsizes[blobno] != -1 &&
+                        blobsizes[blobno] != -2) {
+                        had_errors = 1;
+                        par->verify_status = 1;
+                        locprint(par->sb, par->lua_callback, par->lua_params,
+                            "!%016llx no blob %d found expected sz %d\n",
+                            genid_flipped, blobno, blobsizes[blobno]);
+                    }
+                } else if (rc) {
+                    had_irrecoverable_errors = 1;
+                    par->verify_status = 1;
+                    locprint(par->sb, par->lua_callback, par->lua_params, "!%016llx blob %d rc %d\n",
+                                genid_flipped, blobno, rc);
+                    had_errors = 1;
+                }
+
+                if (rc == 0) {
+                    realblobsz[blobno] = dbt_blob_data.size;
+                    if (blobsizes[blobno] == -1) {
+                        par->verify_status = 1;
+                        locprint(par->sb, par->lua_callback, par->lua_params,
+                            "!%016llx blob %d null but found blob\n",
+                            genid_flipped, blobno);
+                        had_errors = 1;
+                    } else if (blobsizes[blobno] == -2) {
+                        par->verify_status = 1;
+                        locprint(par->sb, par->lua_callback, par->lua_params, "!%016llx blob %d size %d expected "
+                                        "none (inline vutf8)\n",
+                                    genid_flipped, blobno,
+                                    realblobsz[blobno]);
+                        had_errors = 1;
+                    } else if (blobsizes[blobno] != -1 &&
+                               dbt_blob_data.size != blobsizes[blobno]) {
+                        par->verify_status = 1;
+                        locprint(par->sb, par->lua_callback, par->lua_params, "!%016llx blob %d size mismatch "
+                                        "got %d expected %d\n",
+                                    genid_flipped, blobno,
+                                    dbt_blob_data.size, blobsizes[blobno]);
+                        had_errors = 1;
+                    }
+
+                    if (blobsizes[blobno] >= 0 && realblobsz[blobno] >= 0) {
+                        rc = par->add_blob_buffer_callback(
+                            blob_buf, dbt_blob_data.data,
+                            dbt_blob_data.size, blobno);
+                        if (rc)
+                            goto err;
+                    }
+
+                    if (dbt_blob_data.data && had_errors == 0)
```

Comment:
This may be a much older bug, but we always want to free.

Reply:
ok. fixing.